lib/puppet/provider/package/zypper.rb

```diff
@@ -123,4 +123,22 @@ def update
     # zypper install can be used for update, too
     self.install
   end
+
+  def uninstall
+    #extract version numbers and convert to integers
+    major, minor, patch = zypper_version.scan(/\d+/).map{ |x| x.to_i }
+
+    if major < 1
+      super
+    else
+      options = [:remove, '--no-confirm']
+      if major == 1 && minor < 6
+          options << '--force-resolution'
+      end
+
+      options << @resource[:name]
+    end
+
+    zypper *options
```

Comment:
@abednarik doh! One last thing (my fault!) This `zypper` call should go in the `else` clause after `@resource[:name]` is added to `options`, otherwise it could be called even if we call `super`.

After that change, we're ready to merge :)